### ISSUE TITLE
PRO: required checkboxes not validated in multistep

### ISSUE THREAD

**Issue by pwkip** [pwkip/contact-form-7-conditional-fields#55]
https://conditional-fields-cf7.bdwm.be/form-tester/?formcode=%5Bstep%5D+%5Bcheckbox*+mycheckboxes+use_label_element+%221%22+%222%22+%223%22%5D+%5Bstep%5D%0D%0A%5Bsubmit%5D&mailbody=&conditions=

**Comment by pwkip** [comment]
Suggestion from a helpful user. Thank you J. !

> I found why. Explanation and temporary fix proposal.
> 
> // wp-content/plugins/contact-form-7-conditional-fields-pro/pro/step.php
> // 245 : function skip_validation_for_hidden_steps($result, $tags)
> // 259 : if (in_array($invalid_field_key, $this->stepStatus->fieldsInCurrentStep))
> HERE :
> $invalid_field_key = “mycheckboxes”
> $this->stepStatus->fieldsInCurrentStep  = [“mycheckboxes[]”,”mycheckboxes[]”,”mycheckboxes[]”,”mycheckboxes[]”,”mycheckboxes[]”] You can note that because this is multichoice field, its field_key is postfixed with “[]” fieldsInCurrentStep.
> So here, the condition is not performed because of this, and the invalid_field passes through the validation.
> 
> TEMPORARY PROPOSAL FIX
> Change the condition to :
> 
> if (in_array($invalid_field_key, $this->stepStatus->fieldsInCurrentStep) || in_array($invalid_field_key . “[]”, $this->stepStatus->fieldsInCurrentStep))
> IT IS UGLY !! I’m sure you will find nicer solution. (suggestion : valid fieldsInCurrentStep on setter by removing ‘[]’ AND apply array_unique()).

**Comment by JoKolov** [comment]
Suggestion update :

only change function "init_steps" as follow in step.php
```
function init_steps() {
	$this->stepStatus = [];
	if ( isset($_POST['_wpcf7cf_steps']) ) {
		
		$step_data = json_decode(stripslashes($_POST['_wpcf7cf_steps']));

		if (isset($step_data->fieldsInCurrentStep)) {
			$fieldsInCurrentStep = [];
			// remove '[]' from fieldnames to ensure validation
			// '[]' was set for multiple choices fields
			foreach ($step_data->fieldsInCurrentStep as $fieldname) {
				$fieldname = str_replace(['[', ']'], '', $fieldname);
				if (! in_array($fieldname, $fieldsInCurrentStep)) {
					$fieldsInCurrentStep[] = $fieldname;
				}
			}
			$step_data->fieldsInCurrentStep = $fieldsInCurrentStep;
		}

		$this->stepStatus = $step_data;

	}
}
```

**Comment by pwkip** [comment]
fixed in version 1.9
